File: taskflow-frontend/src/types/Category.ts
export interface Category {
    id: number;
    name: string;
    color: string;
    description?: string;
    createdAt: string;
    updatedAt: string;
}

export interface CreateCategoryRequest {
    name: string;
    color: string;
    description?: string;
}

export interface UpdateCategoryRequest {
    name?: string;
    color?: string;
    description?: string;
}

File: TaskFlow.Api/Models/Category.cs
namespace TaskFlow.Api.Models
{
    public class Category
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Color { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public virtual ICollection<Task> Tasks { get; set; } = new List<Task>();
    }
}
